src/tic_tac_toe/core.clj

```diff
@@ -1,56 +1,54 @@
            [tic-tac-toe.protocol.display :refer [display-state display-winner]]))
 
 (def marker-one "x")
 (def marker-two "o")
-
-(defn game-runner [game display player1 player2]
-  (display-state display (:board game))
-  (if-not (winner? (:board game))
-    (let [current-state (next-move player1 game)]
-      (display-state display (:board current-state))
-      (if-not (winner? (:board current-state))
-        (game-runner (next-move player2 current-state) display player1 player2)
-        (display-winner display  (:board current-state))))
-    (display-winner display (:board game))))
-
-(defn opposite-marker [marker]
-  (if (= marker-one marker)
-    marker-two
-    marker-one))
-
-(defmulti create-game (fn [game-type input display board] game-type))
-
-(defmethod create-game :computer-vs-human [game-type-type input display board ]
-  (let [marker-selection (get-marker input)]
-    (if (= marker-selection "x")
-      (let [player-1 (->HumanPlayer "x" input)
-            player-2 (->AiPlayer "o")
-            game {:board board :ai-marker (opposite-marker marker-selection) :player-marker marker-selection}]
-        (game-runner game display player-1 player-2))
-      (let [player-2 (->HumanPlayer "o" input)
-            player-1 (->AiPlayer "x")
-            game {:board board :ai-marker (opposite-marker marker-selection) :player-marker marker-selection}]
-        (game-runner game display player-1 player-2)))))
-
-(defn game-intializer [display input game-type]
-  (let [game game-type
-        board-diemnson (get-board-size input)
-        board (create-empty-board board-diemnson)]
-    (create-game game input display board)))
+(def games [:human-vs-computer :computer-vs-human])
+
+(defn game-runner [game display]
+  (let [game-state (first game)
+        players    (last game )]
+    (display-state display (:board game-state))
+    (if-not (winner? (:board game-state))
+      (let [current-state (next-move (first players) game-state)]
+        (display-state display (:board current-state))
+        (if-not (winner? (:board current-state))
+          (game-runner [current-state (rotate players 0 1)] display)
+          (display-winner display  (:board current-state)))))))3
+
+(defmulti create-game (fn [game-type input board ] game-type))
+
+(defmethod create-game :human-vs-computer [game-type input board]
+  (let [player-1 (->HumanPlayer marker-one input)
+        player-2 (->AiPlayer marker-two)
+        game-state {:board board :ai-marker marker-two :player-marker marker-one}]
+   [game-state [player-1 player-2]]))
+
+(defmethod create-game :computer-vs-human [game-type input board]
+  (let [player-1 (->AiPlayer marker-two)
+        player-2 (->HumanPlayer marker-one input)
+        game-state {:board board :ai-marker marker-one :player-marker marker-two}]
+    [game-state [player-1 player-2]]))
+
+(defn game-intializer [display input]
+  (let [play-type (games (get-game-type input games))
```

Comment:
`play-type` is kind of a weird name to describe the type of game you wish to initialize.